coordinator-tool/src/bin/ec.rs

```diff
@@ -0,0 +1,895 @@
+    election_record::PreVotingData,
+    guardian_public_key::GuardianPublicKey,
+    joint_election_public_key::Ciphertext,
+    verifiable_decryption::{
+        DecryptionProofResponseShare, DecryptionShareResult, VerifiableDecryption,
+    },
+};
+use futures::TryStreamExt;
+use indicatif::{ProgressBar, ProgressStyle};
+use std::collections::{BTreeMap, BTreeSet};
+
+/// Command line configuration of the application.
+#[derive(Debug, clap::Parser)]
+#[command(author, version, about)]
+struct Args {
+    /// The node used for querying
+    #[arg(
+        long = "node",
+        help = "The node endpoint.",
+        default_value = "http://localhost:20001",
+        global = true
+    )]
+    node_endpoint: concordium_rust_sdk::v2::Endpoint,
+    #[command(subcommand)]
+    command:       Command,
+}
+
+#[derive(Debug, clap::Subcommand)]
+enum Command {
+    /// For each account compute the average amount of CCD held
+    /// during the period.
+    #[command(name = "initial-weights")]
+    InitialWeights(RangeWithOutput),
+    /// Look for delegations of the vote during the election period.
+    #[command(name = "final-weights")]
+    FinalWeights {
+        #[arg(long = "out", help = "File to output data into.")]
+        out:             Option<std::path::PathBuf>,
+        #[arg(long = "contract", help = "Address of the election contract.")]
+        contract:        ContractAddress,
+        #[arg(long = "initial-weights", help = "The CSV file with initial weights.")]
+        initial_weights: std::path::PathBuf,
+        #[arg(
+            long = "final-weights",
+            help = "Location where to write the final weights."
+        )]
+        final_weights:   std::path::PathBuf,
+    },
+    /// Tally all the votes.
+    #[command(name = "tally")]
+    Tally(#[clap(flatten)] TallyArgs),
+    /// Compute and optionally post the final result of the election in the
+    /// contract.
+    FinalResult {
+        #[arg(long = "contract", help = "Address of the election contract.")]
+        contract:    ContractAddress,
+        #[arg(
+            long = "admin-keys",
+            help = "Location of the keys used to register election results in the contract."
+        )]
+        wallet_path: Option<std::path::PathBuf>,
+    },
+}
+
+#[derive(Debug, Parser)]
+struct TallyArgs {
+    #[arg(long = "contract", help = "Address of the election contract.")]
+    target_address: ContractAddress,
+    #[arg(
+        long = "final-weights",
+        help = "Location of the file with final weights of accounts."
+    )]
+    final_weights:  std::path::PathBuf,
+    #[arg(
+        long = "admin-keys",
+        help = "Location of the keys used to register election results in the contract."
+    )]
+    keys:           Option<std::path::PathBuf>,
+}
+
+#[derive(Debug, clap::Parser)]
+struct RangeWithOutput {
+    #[arg(
+        long = "start",
+        help = "The start time of the collection. This is inclusive."
```

Comment:
I think it would be good to include a description of how the date should be specified, i.e. 
```suggestion
        help = "The start time of the collection declared as ISO-8061 datetime (e.g. "2012-12-12T12:12:12Z"). This is inclusive."
```